lib/commands/command.bash

```diff
@@ -8,17 +8,49 @@ if [ -z "$(declare -f -F watch_file)" ]; then
   eval "$(asdf exec direnv stdlib)"
 fi
 
+# This is inspired by https://stackoverflow.com/a/1116890
+_follow_symlink() {
+  path="$1"
+
+  # Start in the directory of the (possible) symlink.
+  cd "$(dirname "$path")"
+  filename="$(basename "$path")"
+
+  # Follow symlinks until we run out of symlinks.
+  # This probably will loop forever if there's a cycle.
+  while [ -L "$path" ]; do
+    path="$(readlink "$filename")"
+    cd "$(dirname "$path")"
+    filename="$(basename "$path")"
+  done
+
+  # Now print out the final directory we ended up in, plus the final filename.
+  echo "$(pwd -P)/$filename"
+}
+
 _load_asdf_utils() {
   if [ -z "$(declare -f -F with_plugin_env)" ]; then
-    ASDF_DIR="${ASDF_DIR:-"$(command -v asdf | xargs dirname | xargs dirname)"}"
-    # shellcheck disable=SC1090 # Can't follow non-constant source. Use a directive to specify location.
-    if [ -f "$ASDF_DIR/lib/utils.sh" ]; then
-      # shellcheck disable=SC1091
-      source "$ASDF_DIR/lib/utils.sh"
-    else
-      # shellcheck disable=SC1091
-      source "$ASDF_DIR/lib/utils.bash"
-    fi
+    ASDF_DIR="${ASDF_DIR:-"$(_follow_symlink "$(command -v asdf)" | xargs dirname | xargs dirname)"}"
```

Comment:
I've not been using asdf for about 1.5 years now, haha but when I originally started asdf-direnv, I believe I also added an `asdf env` command on asdf itself, so now I'm wondering if we could just do something like:

`ASDF_DIR="$(asdf env 'echo $ASDF_DIR')"`

or something along that way, like the current asdf for it's root directory instead of trying to find it by ourselves?

Reply:
Hmm... I'm not super clear on what `asdf env` does, but it doesn't seem to do what you'd like on my laptop:

```
$ asdf env 'echo $ASDF_DIR'
unknown command: echo $ASDF_DIR. Perhaps you have to reshim?
```

I don't see how to leverage this to figure out the current asdf's root directory.

Reply:
Right, `asdf env` expects a shim and gives you a shell with that shim's environment.

But looks like `asdf info` could be of help: 

`asdf info 2>/dev/null | grep ASDF_DIR | cut -d= -f2-`

Reply:
Oho! That is VERY USEFUL.

On my Linux box:

```
$ asdf info 2>/dev/null | grep ASDF_DIR
ASDF_DIR=/home/jeremy/.asdf
```

On my Mac box:

```
$ asdf info 2>/dev/null | grep ASDF_DIR
ASDF_DIR=/opt/homebrew/Cellar/asdf/0.9.0/libexec
```

On both of these machines, `$(asdf info 2>/dev/null | grep ASDF_DIR | cut -d= -f2-)/lib/utils.bash` refers to the correct file!